### ISSUE TITLE
"Line ending logic out of sync" assert when calling JsGetAndClearExceptionWithMetadata

### ISSUE THREAD

**Issue by dennis-yemelyanov** [Microsoft/ChakraCore#5805]
Executing a failing script and getting error details with JsGetAndClearExceptionWithMetadata seems to trigger an assert condition in some cases, apparently when some Unicode characters are used.

Sample script:
```
function main() {
  var x = '€' + test();
}
main();
```
Test program (slightly modified version of the C++ HelloWorld sample):
```
#define _JSRT_
#include "ChakraCore.h"
#include <string>
#include <iostream>

using namespace std;

int main()
{
	JsRuntimeHandle runtime;
	JsContextRef context;
	JsValueRef result;
	unsigned currentSourceContext = 0;

	wstring script = L"function main() {\n  var x = '€' + test();\n}\nmain();";

	// Create a runtime. 
	JsCreateRuntime(JsRuntimeAttributeNone, nullptr, &runtime);

	// Create an execution context. 
	JsCreateContext(runtime, &context);

	// Now set the execution context as being the current one on this thread.
	JsSetCurrentContext(context);

	// Run the script.
	JsErrorCode runResult = JsRunScript(script.c_str(), currentSourceContext++, L"", &result);

	if (runResult != JsNoError)
	{
		JsValueRef exception;

		if (JsGetAndClearExceptionWithMetadata(&exception) != JsNoError)
		{
			std::cout << "failed to get exception";
		}
		else
		{
			std::cout << "got exception";
		}
	}
	else
	{
		std::cout << "no error!";
	}

	// Convert your script result to String in JavaScript; redundant if your script returns a String
	JsValueRef resultJSString;
	JsConvertValueToString(result, &resultJSString);

	// Project script result in JS back to C++.
	const wchar_t *resultWC;
	size_t stringLength;
	JsStringToPointer(resultJSString, &resultWC, &stringLength);

	wstring resultW(resultWC);
	cout << string(resultW.begin(), resultW.end()) << endl;
	system("pause");

	// Dispose runtime
	JsSetCurrentContext(JS_INVALID_REFERENCE);
	JsDisposeRuntime(runtime);

	return 0;
}
```

When running this code in debug mode it fails at this assert: https://github.com/Microsoft/ChakraCore/blob/6d96d5b98638fe4535bb4db906378938338050bb/lib/Runtime/Library/JavascriptExceptionMetadata.cpp#L146

After replacing the '€' with something else (for example, 'a'), it works as expected (prints "got exception").

**Comment by dennis-yemelyanov** [comment]
Has this fix been released? The issue seems to still happen in 1.11.7

**Comment by MikeHolman** [comment]
The fix didn't go into 1.11 branch. It will go into 1.12 when that is released.
